### PR TITLE
Add modeled 'TE' header

### PR DESCRIPTION
**Pull request by raboof**
https://tools.ietf.org/html/rfc7230#section-4.3

### CODE REVIEW

**Comment by ignasi35** [comment]
TIL: TE was HTTP/1.1

**Review by ignasi35 — APPROVED** [review]
LGTM

**Review by johanandren — APPROVED** [review]

File: akka-http-core/src/test/scala/akka/http/scaladsl/model/headers/HeaderSpec.scala [hunk 1]
```diff
@@ -154,6 +154,7 @@ class HeaderSpec extends AnyFreeSpec with Matchers {
         `Sec-WebSocket-Protocol`(Vector("chat", "superchat")),
         `Sec-WebSocket-Key`("dGhlIHNhbXBsZSBub25jZQ"),
         `Sec-WebSocket-Version`(Vector(13)),
+        TE(TransferEncodings.trailers),
```
Comment (jrudolph) [inline akka-http-core/src/test/scala/akka/http/scaladsl/model/headers/HeaderSpec.scala:157]: Also needs tests in HttpHeaderSpec (also for the `trailers` value).

**Review by jrudolph — APPROVED** [review]
LGTM but needs a few more tests

**Comment by raboof** [comment]
(will follow up)

**Pull request merged**